platform/ios/src/MGLMapView.mm

```diff
@@ -2727,6 +2727,11 @@ - (void)removeStyleClass:(NSString *)styleClass
     {
         return pair.second.annotation;
     });
+    
+    annotations.erase(std::remove_if(annotations.begin(), annotations.end(),
+                                     [](const id <MGLAnnotation> annotation) { return annotation == nullptr; }),
+                      annotations.end());
+    
     return [NSArray arrayWithObjects:&annotations[0] count:annotations.size()];
```

Comment:
@1ec5 @friedbunny please note this fix for a crash I noticed that I think is already possible in master. Repro is to use iosapp, show the callout on the userdot, then add 100 annotations (which calls `annotations`). Although I think this fix
may be worth keeping, I wanted to call it out since I was thinking it might illustrate a slightly deeper issue (i.e. weird representation of userdot in the _annotationContextsByAnnotationTag map).